### PR TITLE
RHDEVDOCS-4958: Updating the matrix table to fill in identified gaps

### PR DESCRIPTION
**Pull request by shipsing**
**Aligned team**: Dev Tools

**Purpose**: To resolve the following issues: 
https://issues.redhat.com/browse/RHDEVDOCS-4958
https://issues.redhat.com/browse/GITOPS-1776

**OCP version this PR applies to**:  enterprise 4.9 and later

**Link to docs preview**:  https://56052--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/gitops-release-notes.html#GitOps-compatibility-support-matrix_gitops-release-notes

**Change Management**
Need an ACK from Eng:  @jaideepr97 @iam-veeramalla 
Need an ACK from PM: @harrietgrace @wtam2018 
Need an ACK from Product experience: @RickJWagner 
Need an ACK from QE: @varshab1210 
Need an ACK from DPM: @pmkovar 
Need an ACK from CS: @Preeticp 

Peer review: @Srivaralakshmi

### CODE REVIEW

File: modules/go-compatibility-and-support-matrix.adoc [hunk 1]
```diff
@@ -15,12 +15,13 @@ In the table, features are marked with the following statuses:
...
 
 |*Version* |*kam*    |*Helm*  |*Kustomize* |*Argo CD*|*ApplicationSet* |*Dex*     |*RH SSO* |*Notifications Controller*|
-|1.7.0    |0.0.46 TP |3.10.0 GA|4.5.7 GA   |2.5.4 GA |2.4.5 GA       |2.35.1 GA |7.5.1 GA |2.4.5 TP|4.8-4.11
-|1.6.0    |0.0.46 TP |3.8.1 GA|4.4.1 GA   |2.4.5 GA |2.4.5 GA       |2.30.3 GA |7.5.1 GA |2.4.5 TP|4.8-4.10
-|1.5.0    |0.0.42 TP|3.8.0 GA|4.4.1 GA   |2.3.3 GA |0.4.1 TP        |2.30.3 GA |7.5.1 GA ||4.8-4.10
-|1.4.0    |0.0.41 TP|3.7.1 GA|4.2.0 GA   |2.2.2 GA |0.2.0 TP        |2.30.0 GA |7.4.0 GA ||4.7-4.9
-|1.3.0    |0.0.40 TP|3.6.0 GA|4.2.0 GA   |2.1.2 GA |0.2.0 TP        |2.28.0 GA |7.4.0 GA ||4.7-4.9
-
+|1.7.0    |0.0.46 TP |3.10.0 GA|4.5.7 GA   |2.5.4 GA |2.5.4 GA       |2.35.1 GA |7.5.1 GA |2.4.5 TP|4.8-4.11
+|1.6.0    |0.0.46 TP |3.8.1 GA|4.4.1 GA   |2.5.4 GA |2.5.4 GA       |2.30.3 GA |7.5.1 GA |2.4.5 TP|4.8-4.11
```
Comment (varshab1210) [inline modules/go-compatibility-and-support-matrix.adoc:19]: ArgoCD version for GitOps 1.6.0 is 2.4.5 and I think ApplicationSet should be 2.4.5 too

**Comment by shipsing** [comment]
@jaideepr97 @iam-veeramalla @harrietgrace @wtam2018 @RickJWagner@varshab1210, please review this PR and share your feedback. Thanks

**Comment by harrietgrace** [comment]
Looks great, thank you!

**Comment by Srivaralakshmi** [comment]
@shipsing looks great! Thanks for all the hard work, LGTM.

@harrietgrace Thanks for the support and co-operation :)

**Comment by RickJWagner** [comment]
Thank you for providing the preview link, this is very useful.
https://56052--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/gitops-release-notes.html#GitOps-compatibility-support-matrix_gitops-release-notes

PLM finds this PR to be good.

**Comment by wtam2018** [comment]
The "Table 1. Technology Preview tracker" needs to be updated to add OCP 4.12.

**Comment by shipsing** [comment]
@wtam2018, Incorporated your suggestion. Please verify and confirm. Thanks

**Comment by Preeticp** [comment]
Cs LGTM

**Pull request merged**